### PR TITLE
[Fabric] Shim UIViewContentMode

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [x] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Fabric on macOS implementation:
Shim UIViewContentMode enum and disable handleContentSizeCategoryDidChangeNotification

## Changelog

[macOS][Fabric] - Shim UIViewContentMode

## Test Plan

[x] Build RNTester-macOS w/ Fabric - doesn’t run yet, but no UIViewContentMode errors
CleanShot 2022-12-12 at 17 10 41

Build errors: 
[Build RNTester-macOS_2022-12-12T17-09-55.txt](https://github.com/microsoft/react-native-macos/files/10213397/Build.RNTester-macOS_2022-12-12T17-09-55.txt)

[x] Build RNTester - iOS w/ Fabric
CleanShot 2022-12-12 at 17 11 41

[x] Build RNTester-macOS w/ Paper - should work
CleanShot 2022-12-12 at 17 20 31

[x] Build RNTester - iOS w/ Paper - should work
CleanShot 2022-12-12 at 17 18 55

### CODE REVIEW

**Comment by shwanton** [comment]
closes #1556

File: React/Base/RCTUIKit.h [hunk 1]
```diff
@@ -206,6 +206,7 @@ enum : NSUInteger
 };
 
 // UIView/NSView.h
+#define UIViewContentMode NSInteger // TODO(macOS GH#774)
 enum : NSInteger
```
Comment (Saadnajmi) [inline React/Base/RCTUIKit.h:210]: Should this be something more like 
```
typedef NS_ENUM(NSInteger, UIViewContentMode) {
...
}

File: React/Fabric/Mounting/ComponentViews/Text/RCTParagraphComponentView.mm [hunk 2]
```diff
@@ -42,7 +42,9 @@ - (instancetype)initWithFrame:(CGRect)frame
     _props = defaultProps;
 
     self.opaque = NO;
+#if !TARGET_OS_OSX // [TODO(macOS GH#774)
```
Comment (Saadnajmi) [inline React/Fabric/Mounting/ComponentViews/Text/RCTParagraphComponentView.mm:45]: Comment about block tags here

**Review by Saadnajmi — COMMENTED** [review]
Mostly good, save a few comments!

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**